parser/src/parser/tests/mod.rs

```diff
@@ -1,18 +1,619 @@
+            $(miden_diagnostics::Span::new(miden_diagnostics::SourceSpan::UNKNOWN, (ident!($binding_name), $binding_size))),*
+        ])
+    }
+}
+
+macro_rules! random_values {
+    ($name:literal, $size:literal) => {
+        RandomValues::with_size(miden_diagnostics::SourceSpan::UNKNOWN, ident!($name), $size)
+    };
+
+    ($name:literal, [$(($binding_name:ident, $binding_size:literal)),*]) => {
+        RandomValues::new(miden_diagnostics::SourceSpan::UNKNOWN, ident!($name), vec![
+            $(miden_diagnostics::Span::new(miden_diagnostics::SourceSpan::UNKNOWN, (ident!($binding_name), $binding_size))),*
+        ])
+    }
+}
+
+macro_rules! constant {
+    ($name:ident = $value:literal) => {
+        Constant::new(
+            SourceSpan::UNKNOWN,
+            ident!($name),
+            ConstantExpr::Scalar($value),
+        )
+    };
+
+    ($name:ident = [$($value:literal),+]) => {
+        Constant::new(SourceSpan::UNKNOWN, ident!($name), ConstantExpr::Vector(vec![$($value),+]))
+    };
+
+    ($name:ident = [$([$($value:literal),+]),+]) => {
+        Constant::new(SourceSpan::UNKNOWN, ident!($name), ConstantExpr::Matrix(vec![$(vec![$($value),+]),+]))
+    };
+}
+
+macro_rules! vector {
+    ($($value:expr),*) => {
+        Expr::Vector(miden_diagnostics::Span::new(miden_diagnostics::SourceSpan::UNKNOWN, vec![$($value),*]))
+    }
+}
+
+macro_rules! matrix {
+    ($([$($value:expr),+]),+) => {
+        Expr::Matrix(miden_diagnostics::Span::new(miden_diagnostics::SourceSpan::UNKNOWN, vec![$(vec![$($value),+]),+]))
+    };
+}
+
+macro_rules! let_ {
+    ($name:ident = $value:expr => $($body:expr),+) => {
+        Statement::Let(Let::new(miden_diagnostics::SourceSpan::UNKNOWN, ident!($name), $value, vec![$($body),+]))
+    };
+}
+
+macro_rules! enforce {
+    ($expr:expr) => {
+        Statement::Enforce($expr)
+    };
+}
+
+macro_rules! enforce_all {
+    ($expr:expr) => {
+        Statement::EnforceAll($expr)
+    };
+}
+
+macro_rules! lc {
+    (($(($binding:ident, $iterable:expr)),+) => $body:expr) => {{
+        let context = vec![
+            $(
+                (ident!($binding), $iterable)
+            ),+
+        ];
+        ListComprehension::new(miden_diagnostics::SourceSpan::UNKNOWN, $body, context, None)
+    }};
+
+    (($(($binding:literal, $iterable:expr)),+) => $body:expr) => {{
+        let context = vec![
+            $(
+                (ident!($binding), $iterable)
+            ),+
+        ];
+        ListComprehension::new(miden_diagnostics::SourceSpan::UNKNOWN, $body, context, None)
+    }};
+
+    (($(($binding:ident, $iterable:expr)),*) => $body:expr, when $selector:expr) => {{
+        let context = vec![
+            $(
+                (ident!($binding), $iterable)
+            ),+
+        ];
+        ListComprehension::new(miden_diagnostics::SourceSpan::UNKNOWN, $body, context, Some($selector))
+    }};
+
+    (($(($binding:literal, $iterable:expr)),*) => $body:expr, when $selector:expr) => {{
```

Comment:
this is such a tiny nit, but I would probably drop the comma before `when`. It doesn't mirror the language, so I don't feel like it's helpful, and I think it would make me likely to make an error when using the macro. It's not a big deal though

Reply:
I absolutely would have done that (and in fact attempted to initially), but the `expr` pattern type can only be followed by a handful of tokens, and `when` is not one of those. I decided to separate the selector into a separate argument, but kept the `when` thinking it would be easier to read the macro with it present than with only a comma.

Procedural macros could work magic here, but that has a much higher implementation cost which wasn't worth it in my opinion

Reply:
Yep, makes sense. I wondered if that might be the case